modules/core/src/lib/deck.ts

```diff
@@ -477,6 +477,25 @@ export default class Deck<ViewsT extends ViewOrViews = null> {
       viewState: this._getViewState()
     });
 
+    if (props.device && props.device.id !== this.device?.id) {
```

Comment:
Are we just trying to determine if there is a new device?
```suggestion
    if (props.device && props.device !== this.device) {
```

TBH, I did not expect that deck would support reinitializing with a new device. I think that is going to have a long list of nasty corner cases.

I would create a new deck instance when we switch device.

Reply:
Ideally deck can handle a device change for the user to keep the API declarative. I agree there is probably long tail of edges cases preventing reuse, but if that's what the user instructs deck to do, then I believe it should do whatever it needs to do to make it happen... even if it means a full-reinitialization.

Reply:
yeah effectively this is just checking if the device has been re-created as a new device. given that deck accepts a device prop, it feels a bit weird to not handle the case where that changes. i agree it's probably a very small minority of users that might want to do it but this appears to be all the re-initialisation we need to do.